diff -up cups-1.4.2/scheduler/ipp.c.format_log-segfault cups-1.4.2/scheduler/ipp.c
--- cups-1.4.2/scheduler/ipp.c.format_log-segfault	2014-04-02 12:38:55.053777366 +0100
+++ cups-1.4.2/scheduler/ipp.c	2014-04-02 12:39:21.495104286 +0100
@@ -10360,8 +10360,9 @@ send_document(cupsd_client_t  *con,	/* I
     if (!filetype)
       filetype = mimeType(MimeDatabase, super, type);
 
-    cupsdLogJob(job, CUPSD_LOG_DEBUG, "Request file type is %s/%s.",
-		filetype->super, filetype->type);
+    if (filetype)
+      cupsdLogJob(job, CUPSD_LOG_DEBUG, "Request file type is %s/%s.",
+		  filetype->super, filetype->type);
   }
   else
     filetype = mimeType(MimeDatabase, super, type);
