diff -up cups-1.4.2/backend/snmp-supplies.c.str3809 cups-1.4.2/backend/snmp-supplies.c
--- cups-1.4.2/backend/snmp-supplies.c.str3809	2011-06-16 15:45:26.685605440 +0200
+++ cups-1.4.2/backend/snmp-supplies.c	2011-06-16 15:46:19.750373804 +0200
@@ -401,9 +401,9 @@ backend_init_supplies(
   * See if we should be getting supply levels via SNMP...
   */
 
-  if ((ppd = ppdOpenFile(getenv("PPD"))) != NULL &&
-      (ppdattr = ppdFindAttr(ppd, "cupsSNMPSupplies", NULL)) != NULL &&
-      ppdattr->value && strcasecmp(ppdattr->value, "true"))
+  if ((ppd = ppdOpenFile(getenv("PPD"))) == NULL ||
+      ((ppdattr = ppdFindAttr(ppd, "cupsSNMPSupplies", NULL)) != NULL &&
+       ppdattr->value && strcasecmp(ppdattr->value, "true")))
   {
     ppdClose(ppd);
     return;
