diff -up cups-1.4.2/man/cupsd.conf.man.in.str4457 cups-1.4.2/man/cupsd.conf.man.in
--- cups-1.4.2/man/cupsd.conf.man.in.str4457	2015-02-23 18:10:10.758093467 +0000
+++ cups-1.4.2/man/cupsd.conf.man.in	2015-02-23 18:11:07.486868383 +0000
@@ -348,6 +348,18 @@ FileDevice No
 Specifies whether the file pseudo-device can be used for new
 printer queues.
 .TP 5
+\fBErrorPolicy abort-job\fR
+Specifies that a failed print job should be aborted (discarded) unless otherwise specified for the printer.
+.TP 5
+\fBErrorPolicy retry-job\fR
+Specifies that a failed print job should be retried at a later time unless otherwise specified for the printer.
+.TP 5
+\fBErrorPolicy retry-this-job\fR
+Specifies that a failed print job should be retried immediately unless otherwise specified for the printer.
+.TP 5
+\fBErrorPolicy stop-printer\fR
+Specifies that a failed print job should stop the printer unless otherwise specified for the printer. The 'stop-printer' error policy is the default.
+.TP 5
 FilterLimit limit
 .br
 Specifies the maximum cost of filters that are run concurrently.
