diff -up cups-1.4.2/cups/request.c.ipp-consume-cpu cups-1.4.2/cups/request.c
--- cups-1.4.2/cups/request.c.ipp-consume-cpu	2016-09-22 16:53:14.885803305 +0200
+++ cups-1.4.2/cups/request.c	2016-09-22 16:56:07.609543749 +0200
@@ -266,11 +266,11 @@ cupsDoIORequest(http_t     *http,	/* I -
 
     DEBUG_printf(("2cupsDoIORequest: status=%d", status));
 
-    if (status >= HTTP_BAD_REQUEST &&
-	status != HTTP_UNAUTHORIZED &&
-	status != HTTP_UPGRADE_REQUIRED)
+    if (status == HTTP_ERROR ||
+        (status >= HTTP_BAD_REQUEST &&
+         status != HTTP_UNAUTHORIZED &&
+         status != HTTP_UPGRADE_REQUIRED))
     {
-      httpFlush(http);
       _cupsSetHTTPError(status);
       break;
     }
