diff -up cups-1.4.2/scheduler/client.c.nodatadelay cups-1.4.2/scheduler/client.c
--- cups-1.4.2/scheduler/client.c.nodatadelay	2014-06-16 15:44:52.043739500 +0100
+++ cups-1.4.2/scheduler/client.c	2014-06-16 15:46:22.825193077 +0100
@@ -2104,7 +2104,9 @@ cupsdReadClient(cupsd_client_t *con)	/*
 
 	  if (con->http.state != HTTP_POST_SEND)
 	  {
-            if ((bytes = httpRead2(HTTP(con), line, sizeof(line))) < 0)
+	    if (!httpWait(HTTP(con), 0))
+	      return;
+            else if ((bytes = httpRead2(HTTP(con), line, sizeof(line))) < 0)
 	    {
 	      if (con->http.error && con->http.error != EPIPE)
 		cupsdLogMessage(CUPSD_LOG_DEBUG,
