diff -up cups-1.4.2/filter/raster.c.str4551 cups-1.4.2/filter/raster.c
--- cups-1.4.2/filter/raster.c.str4551	2008-07-12 00:46:21.000000000 +0200
+++ cups-1.4.2/filter/raster.c	2015-06-10 18:30:20.150230785 +0200
@@ -239,8 +239,11 @@ cupsRasterReadHeader(
   */
 
   if (!cups_raster_read_header(r))
+  {
+    memset(h, 0, sizeof(cups_page_header_t));
     return (0);
-  
+  }
+
  /*
   * Copy the header to the user-supplied buffer...
   */
@@ -268,9 +271,12 @@ cupsRasterReadHeader2(
   */
 
   if (!cups_raster_read_header(r))
-    return (0);
-  
- /*
+  {
+    memset(h, 0, sizeof(cups_page_header2_t));
+     return (0);
+  }
+
+/*
   * Copy the header to the user-supplied buffer...
   */
 
@@ -302,7 +308,7 @@ cupsRasterReadPixels(cups_raster_t *r,	/
   int		count;			/* Repetition count */
 
 
-  if (r == NULL || r->mode != CUPS_RASTER_READ || r->remaining == 0)
+  if (r == NULL || r->mode != CUPS_RASTER_READ || r->remaining == 0 || r->header.cupsBytesPerLine == 0)
     return (0);
 
   if (!r->compressed)
@@ -756,7 +762,7 @@ cups_raster_read_header(
 
   cups_raster_update(r);
 
-  return (1);
+  return (r->header.cupsBytesPerLine != 0 && r->header.cupsHeight != 0 && (r->header.cupsBytesPerLine % r->bpp) == 0);
 }
 
 
